cfgrib/dataset.py

```diff
@@ -521,13 +521,18 @@ def build_variable_components(
     extra_coords_data: T.Dict[str, T.Dict[str, T.Any]] = {
         coord_name: {} for coord_name in extra_coords
     }
-    for dim in header_dimensions:
-        header_value_index[dim] = {v: i for i, v in enumerate(coord_vars[dim].data.tolist())}
+    extra_dims = list(extra_coords.values())
```

Comment:
If you use `tuple` here you may remove the cast to `list` later. 

```suggestion
    extra_dims = tuple(extra_coords.values())
```